server/src/test/java/org/elasticsearch/env/NodeEnvironmentTests.java

```diff
@@ -211,39 +212,47 @@ public void testResolveIndexFolders() throws Exception {
     public void testDeleteSafe() throws Exception {
         final NodeEnvironment env = newNodeEnvironment();
         final Index index = new Index("foo", "fooUUID");
-        ShardLock fooLock = env.shardLock(new ShardId(index, 0), "1");
+        final ShardLock fooLock = env.shardLock(new ShardId(index, 0), "1");
         assertEquals(new ShardId(index, 0), fooLock.getShardId());
 
         for (Path path : env.indexPaths(index)) {
             Files.createDirectories(path.resolve("0"));
             Files.createDirectories(path.resolve("1"));
         }
 
-        try {
-            env.deleteShardDirectorySafe(new ShardId(index, 0), idxSettings);
-            fail("shard is locked");
-        } catch (ShardLockObtainFailedException ex) {
-            // expected
+        {
+            SetOnce<Path[]> listener = new SetOnce<>();
+            ShardLockObtainFailedException ex = expectThrows(ShardLockObtainFailedException.class,
+                () -> env.deleteShardDirectorySafe(new ShardId(index, 0), idxSettings, listener::set));
+            assertNull(listener.get());
```

Comment:
Maybe `assert false` rather than `listener::set`?

Reply:
Oups, I missed it the last time. I pushed aea4e6db93d580c453eb81694fca2ea42bbaf7b7